src/test/blockmanager_tests.cpp

```diff
@@ -133,11 +132,78 @@ BOOST_FIXTURE_TEST_CASE(blockmanager_block_data_availability, TestChain100Setup)
     func_prune_blocks(last_pruned_block);
 
     // 3) The last block not pruned is in-between upper-block and the genesis block
-    BOOST_CHECK_EQUAL(&blockman.GetFirstBlock(tip, BLOCK_HAVE_DATA), first_available_block);
+    BOOST_CHECK_EQUAL(blockman.GetFirstBlock(tip, BLOCK_HAVE_DATA), first_available_block);
     BOOST_CHECK(blockman.CheckBlockDataAvailability(tip, *first_available_block));
     BOOST_CHECK(!blockman.CheckBlockDataAvailability(tip, *last_pruned_block));
 }
 
+BOOST_FIXTURE_TEST_CASE(blockmanager_block_data_part, TestChain100Setup)
+{
+    LOCK(::cs_main);
+    auto& chainman = m_node.chainman;
+    auto& blockman = chainman->m_blockman;
+    const CBlockIndex& tip = *chainman->ActiveTip();
+    const FlatFilePos tip_block_pos = tip.GetBlockPos();
+
+    std::vector<std::byte> block{};
+    BOOST_CHECK(blockman.ReadRawBlock(block, tip_block_pos));
+    BOOST_CHECK_GE(block.size(), 200);
+
+    std::vector<std::byte> block_part{};
+    const auto read_tip_part{[&](auto part_offset, auto part_size) {
+        block_part.clear();
+        auto block_result{blockman.ReadRawBlock(tip_block_pos, part_offset, part_size)};
+        if (auto* vec{std::get_if<std::vector<std::byte>>(&block_result)}) {
+            block_part = std::move(*vec);
+            return true;
+        }
+        return false;
```

Comment:
Aren't we just copying the implementation from `bool BlockManager::ReadRawBlock(std::vector<std::byte>& block, const FlatFilePos& pos) const` here?